### ISSUE TITLE
Port forwarding unique checking does not use IP address

### ISSUE THREAD

**Issue by andkirby** [mitchellh/vagrant#7905]
In a nutshell, there is no ability to forward the same port of different IP addresses to a guest OS.

E.g.:
Host 127.0.0.111:80 => guest 8011
Host 127.0.0.222:80 => guest 8022
### Vagrant version

1.8.6
### Host operating system

Windows 10
### Guest operating system

CentOS
### Vagrantfile

``` ruby
VAGRANTFILE_API_VERSION = '2'

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = 'centos/7'
  config.vm.box_check_update = false
  config.vm.boot_timeout = 300

  config.vm.network 'forwarded_port', guest: 8011, host: 80, host_ip: '127.0.0.111', id: 'aaa'
  config.vm.network 'forwarded_port', guest: 8022, host: 80, host_ip: '127.0.0.222', id: 'bbb'
end
```
### Expected behavior

Added ports forwarding as expected:
- Host 127.0.0.111:80 => guest 8011
- Host 127.0.0.222:80 => guest 8022
### Actual behavior

Error output

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* Forwarded port '80' (host port) is declared multiple times
with the protocol 'tcp'.
```
### Steps to reproduce
1. Run `vagrant up` in the same directory as `Vagrantfile`
### References

I have found the code which responsible for this:
https://github.com/mitchellh/vagrant/blob/7eb1091a7102dd4d2ebab762ee1658689a889880/plugins/kernel_v2/config/vm.rb#L688

In my understanding there is the `host_ip` option should be used. And, seem, guest_ip (without it, it will produce the same problem but on the guest side.)

**Comment by andkirby** [comment]
Stackoverflow [issue](http://stackoverflow.com/questions/40089125/how-to-solve-overwritten-port-forwarding-in-virtualbox-via-vagrant/40090285#40090285).

**Comment by andkirby** [comment]
P.S.
Seem, the method `VagrantPlugins::Kernel_V2::VMConfig::validate()` is really huge. Can it be refactored?..

**Comment by chrisroberts** [comment]
#7035

**Comment by chrisroberts** [comment]
Fixed via: #7035
